keyutils-raw/src/functions.rs

```diff
@@ -33,111 +33,111 @@ extern "C" {
         description:    *const libc::c_char,
         payload:        *const libc::c_void,
         plen:           libc::size_t,
-        keyring:        key_serial_t)
-        -> key_serial_t;
+        keyring:        KeyringSerial)
+        -> KeyringSerial;
     pub fn request_key(
         type_:          *const libc::c_char,
         description:    *const libc::c_char,
         callout_info:   *const libc::c_char,
-        keyring:        key_serial_t)
-        -> key_serial_t;
+        keyring:        Option<KeyringSerial>)
```

Comment:
I'm seeing this warning. Seems that `repr(C)` is required to make this actually C-compliant.

```
warning: `extern` block uses type `std::option::Option<std::num::NonZeroI32>` which is not FFI-safe: enum has no representation hint
  --> keyutils-raw/src/functions.rs:42:25
   |
42 |         keyring:        Option<KeyringSerial>)
   |                         ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(improper_ctypes)] on by default
   = help: consider adding a #[repr(...)] attribute to this enum
```

Same thing below.

Reply:
So I was seeing that warning as well. Essentially, this warning isn't correct, Rust does guarantee that the representations of the following types are the same:

  - `NonZeroI32`
  - `Option<NonZeroI32>`
  - `i32`

The [linked issue above](https://github.com/rust-lang/rust/pull/60300) talks about FFI safety, and fixes this warning. Note that this warning does not appear in Rust 1.37 (current nightly). I can silence the warning if you want me to.

Reply:
Added `#[allow(improper_ctypes)]` to that file to fix the warning.